app/src/main/java/io/neurolab/communication/USBReceiver.java

```diff
@@ -0,0 +1,69 @@
+package io.neurolab.communication;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.hardware.usb.UsbManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.felhr.usbserial.UsbSerialInterface;
+
+import java.io.UnsupportedEncodingException;
+
+import io.neurolab.utilities.FilePathUtil;
+
+public class USBReceiver extends BroadcastReceiver {
+    private USBCommunicationHandler usbCommunicationHandler;
+
+    private static String dataContent = "";
+    private boolean collecting = false;
+    private static final char endDetectChar= 'R';
+
+    public USBReceiver(USBCommunicationHandler usbCommunicationHandler) {
+        this.usbCommunicationHandler = usbCommunicationHandler;
+    }
+
+    private UsbSerialInterface.UsbReadCallback readCallback = new UsbSerialInterface.UsbReadCallback() { //Defining a Callback which triggers whenever data is read.
+        @Override
+        public void onReceivedData(byte[] arg0) {
+            String data;
+            try {
+                data = new String(arg0, "UTF-8");
+                for (int i = 0; i < data.length(); i++) {
+                    if (data.charAt(i) == endDetectChar) { // To detect the end of file recording
```

Comment:
Why are we using a letter `R` to detect the EOF? Are we adding a `R` to the end of each set?